cmd/terramate/cli/tmcloud/auth/credential.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2025 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package auth
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/golang-jwt/jwt"
+	"github.com/terramate-io/terramate/cloud"
+	"github.com/terramate-io/terramate/cmd/terramate/cli/cliconfig"
+	"github.com/terramate-io/terramate/cmd/terramate/cli/out"
+	"github.com/terramate-io/terramate/errors"
+)
+
+const (
+	defaultCloudTimeout = 60 * time.Second
+
+	// ErrIDPNeedConfirmation is an error indicating the user has multiple providers set up and
+	// linking them is needed.
+	ErrIDPNeedConfirmation errors.Kind = "the account was already set up with another email provider"
+
+	// ErrEmailNotVerified is an error indicating that user's email need to be verified.
+	ErrEmailNotVerified errors.Kind = "email is not verified"
+
+	// ErrLoginRequired is an error indicating that user has to login to the cloud.
+	ErrLoginRequired errors.Kind = "cloud login required"
+)
+
+// Credential is the interface for the cloud credentials.
+type Credential interface {
+	Name() string
```

Comment:
_nitpick_ it would be ideal to put some GoDoc here to all future implementations for guidance if it's used as library now